providers/infoblox/infoblox.go

```diff
@@ -64,7 +72,18 @@ func (d *InfobloxProvider) Init(rootDomainName string) error {
 	}
 
 	if password = os.Getenv("INFOBLOX_PASSWORD"); len(password) == 0 {
-		return fmt.Errorf("INFOBLOX_PASSWORD is not set")
+		if secretFile = os.Getenv("INFOBLOX_SECRET"); len(secretFile) == 0 {
+			return fmt.Errorf("INFOBLOX_PASSWORD nor INFOBLOX_SECRET are not set")
+		}
+
+		// If password nil, using secrets
+		p, err := ioutil.ReadFile(secretFile)
+    	if err != nil {
+        	return fmt.Errorf("Error reading INFOBLOX_SECRET %s: %v", secretFile, err)
+    	}
+
+    	logrus.Debugf("Infoblox using INFOBLOX_SECRET %s", secretFile)
```

Comment:
Nit: should we check if the file is empty? and return an error if so

Reply:
INFOBLOX_SECRET file is to get password from secrets. It's just used if INFOBLOX_PASSWORD is empty. If secret file couldn't be read, return error due to couldn't get password.

Reply:
@rawmind0 I don't know, you tell me :)

Reply:
Adding check if read password is empty.